src/backends/sqlite3/statement.cpp

```diff
@@ -353,11 +350,7 @@ sqlite3_statement_backend::execute(int number)
 statement_backend::exec_fetch_result
 sqlite3_statement_backend::fetch(int number)
 {
-    if (hasVectorIntoElements_ || number == 0)
-        return load_rowset(number);
-    else
-        return load_one();
-
+    return (1 == number) ? load_one() : load_rowset(number);
```

Comment:
It looks like `hasVectorIntoElements_` is not used at all now and should be removed?